packages/core-data/src/entities.js

```diff
@@ -304,6 +304,7 @@ async function loadPostTypeEntities() {
 			baseURLParams: { context: 'edit' },
 			name,
 			label: postType.name,
+			slug: postType.slug,
```

Comment:
What is this change about?

Reply:
The idea was to add the post type slug to the entity object so it is available when calling `getEntitiesConfig( 'postType' )`. If I am not mistaken, it can be different than the name. We are using it to get the slugs of all registered post types.

The slug information is available when calling `wp/v2/types` [here](https://github.com/WordPress/gutenberg/blob/24e0abcd31393fc7e3e9614d2e7646e8136bfd45/packages/core-data/src/entities.js#L294) so, the same way we are adding the `label`, I added the `slug`. Can this become a problem? Is not expected to have the post type slug in the entity config?

Reply:
But this function returns entity configs, not post types, if this was typescript, it would have returned an error. We shouldn't try to force something to be something else, just because it's convenient.

Reply:
Mmm okay. I somehow understood it was specific to post types reading the `loadPostTypeEntitities` function, but I get it now. Thanks for the clarification. I'll look for an alternative way to reuse the preloaded API call.

Reply:
I understand the type problem, but what is the problem to enriching the type with slug? Genuine question, does slug have something which would dillute the entity config type?

Reply:
We're not enhancing all the entity types with "slug", we're enhancing only the "post types" entity types which means we're creating an inconsistency. Also, entity types already have "kind" and "name", what differentiate "slug" from "name" is very unclear.

Reply:
> We're not enhancing all the entity types with "slug", we're enhancing only the "post types" entity types 

Oh indeed, I see! Thank you! 😄

Reply:
It sounds like the solution would be to create a selector and resolver pair that abstracts the following lines:
```js
const postTypes = await apiFetch( {
	path: '/wp/v2/types?context=view',
} );
```
and use the selector in both places rather than calling `getEntitiesConfig` which exposes additional details.

Reply:
I see it's more nuanced, as you can still provide a custom slug for some reasons:

https://developer.wordpress.org/plugins/post-types/registering-custom-post-types/#a-custom-slug-for-a-custom-post-type

I'm not sure what the implications are though.

Reply:
Yes, that's what I tested. But the Template Hierarchy seems to use the `key`. At least is what I understood from the docs and what I saw in my tests.